images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

```diff
@@ -0,0 +1,352 @@
+        eventType = self.params.get("type")
+        source = self.params.get("source")
+
+        definition = defaultdict(defaultdict)
+
+        meta = definition["metadata"]
+        meta["name"] = self.params.get("name")
+        meta["namespace"] = self.params.get("namespace")
+
+        involvedObj = self.params.get("involvedObject")
+        print(involvedObj["kind"],involvedObj["namespace"],involvedObj["name"],involvedObj["apiVersion"])
+
+
+        resource = self.find_resource(
+            "Event", "v1", fail=True)
+
+        priorEvent = None
+        try:
+            priorEvent = resource.get(
+                name=meta["name"],
+                namespace=meta["namespace"])
+
+        except openshift.dynamic.exceptions.NotFoundError:
+            pass
+
+        # event counter
+        priorCount = 1
+        print("the count is no event exists %i" % priorCount)
+        now = datetime.datetime.now()
+        rfc = kubernetes.config.dateutil.format_rfc3339(now)
+        firstTimestamp = rfc
+        lastTimestamp = rfc
+
+        if priorEvent:
+            if priorEvent["reason"] != reason:
+                print("If event exists and reason is changed %i" % priorCount)
+                now = datetime.datetime.now()
+                rfc = kubernetes.config.dateutil.format_rfc3339(now)
+                priorCount = 1
+                firstTimestamp = rfc
+                print("the value of the firstTimestamp new event is", firstTimestamp)
+                lastTimestamp = rfc
+                print("the value of the lastTimestamp new event is", lastTimestamp)
+            else:
+                priorCount = priorEvent["count"] + 1
+                print("If event exists and reason is the same %i" % priorCount)
+                firstTimestamp = priorEvent["firstTimestamp"]
+                print("the value of the firstTimestamp event exists same reason is", firstTimestamp)
+                lastTimestamp = firstTimestamp
+                print("the value of the lastTimestamp event exists same reason is", lastTimestamp)
+
+        # handle involvedObject
+        involvedObjectResourceVersion = None
+        involvedObjectUid = None
+        involvedObjectResource = self.find_resource(
+            involvedObj["kind"], "v1", fail=True)
+
+        if involvedObjectResource:
+            try:
+                apiInvolvedObject = resource.get(
+                    name=involvedObj["name"], namespace=involvedObj["namespace"])
+
+                involvedObjectUid = apiInvolvedObject["metadata"]["uid"]
+                involvedObjectResourceVersion = apiInvolvedObject["metadata"]["resourceVersion"]
+
+            except openshift.dynamic.exceptions.NotFoundError:
+                pass
+
+        event = {
+            "apiVersion": "v1",
+            "count": priorCount,
+            "eventTime": None,
+            "firstTimestamp": firstTimestamp,
+            "involvedObject": {
```

Comment:
Check if you can just pass `"involvedObject": involved_obj` as it looks like this should be possible.

Reply:
I realize that this implies that we need to manually set/update the resource version and uid if the event already exists.

Reply:
Ah yes. Ill make a comment above on that.

Reply:
Yeah change this to `involvedObject": involved_obj` once you've removed `involvedObjectResourceVersion` and `involvedObjectResourceVersion` like I mentioned above.